torch/nn/functional.py

```diff
@@ -2572,6 +2572,24 @@ def layer_norm(
         )
     return torch.layer_norm(input, normalized_shape, weight, bias, eps, torch.backends.cudnn.enabled)
 
+def rms_norm(
+    input: Tensor,
+    normalized_shape: List[int],
+    weight: Optional[Tensor] = None,
+    eps: Optional[float] = None,
+) -> Tensor:
+    r"""Apply Root Mean Square Layer Normalization.
+
+    See :class:`~torch.nn.RMSNorm` for details.
+    """
+    # isintance(input, Tensor) needed for classes that implement __torch_function__ but are not Tensors
+    if eps is None and isinstance(input, Tensor):
```

Comment:
I'm curious why we resolve this here and not after torch_function handling or within the C++ code?